trollflow2/launcher.py

```diff
@@ -84,7 +88,7 @@ def get_test_message(test_message_file):
     return msg
 
 
-def check_results(produced_files, start_time, exitcode):
+def check_results(produced_files, start_time, exitcode, remote_filesystem=None):
```

Comment:
Instead of passing the filesystem here, could we have `produced_files` be a list of FSFiles (or subclass of it) so that it contains both filename and filesystem?
Also, having a subclass of FSFile would allow to add a check_result method to it, which would be nice.

Reply:
This isn't actually the filesystem (like 's3' or any fs object), it's the URI to the S3 bucket. The filenames to `check_results()` are coming from `save_datasets()` plugin, which doesn't know about the `s3_uploader()`. And I think it shouldn't either.

I'll clarify the argument naming.

Reply:
Argument name changed in https://github.com/pytroll/trollflow2/pull/158/commits/27f62d05b100c45d1aec3171cb3fe498a98e4a17